py/embedded-server/deephaven_server/__init__.py

```diff
@@ -14,3 +19,55 @@
 # distribution name from the package name.
 # https://docs.python.org/3/library/importlib.metadata.html#package-distributions
 __version__ = importlib.metadata.version('deephaven-server')
+
+
+def main():
+    """
+    Main entry point for the Deephaven embedded server application.
+    Accepts a command to start the application and the arguments to use.
+    Defaults to `start`, which starts the server.
+    """
+    parser = argparse.ArgumentParser(description="Start the Deephaven embedded server")
+    parser.add_argument("cmd", nargs="?", default="start", choices=["start"], help="Deephaven command to run. Defaults to start.")
+    parser.add_argument("-p", "--port", default=8080, type=int, help="The port to bind to. Defaults to 8080.")
+    parser.add_argument("--host", default="localhost", help="The host to bind to. Defaults to localhost.")
+    parser.add_argument("--key", default=None, help="The key to use. Uses a randomly generated key if not specified.")
+    parser.add_argument("--jvm-args", default=None, help="The JVM arguments to use.")
+    parser.add_argument("--dh-args", default=None, help="The Deephaven arguments to use.")
+    args = parser.parse_args()
```

Comment:
Click provides many useful features.  

Environment variables are just one important one.
https://click.palletsprojects.com/en/8.1.x/options/#values-from-environment-variables
https://click.palletsprojects.com/en/8.1.x/options/#multiple-values-from-environment-values

This could be combined with `.env` support to make a really flexible tool.
https://dev.to/jakewitcher/using-env-files-for-environment-variables-in-python-applications-55a1

Reply:
I'll take a look

Reply:
I don't have a horse in this race; just noting some stuff I learned recently. click is based on optparse, and optparse is soft deprecated. https://www.bitecode.dev/i/135568834/pep-defines-soft-deprecation-getopt-and-optparse-soft-deprecated

This article does specifically point out https://typer.tiangolo.com/ and https://click.palletsprojects.com/en/8.1.x/, so I think it's reasonable to compare argparse vs click vs typer.

Reply:
Yea the nice thing about `argparse` is it's baked in. We're really not doing a lot here, I don't think it really matters which one we use?

Reply:
It is going to matter what we use because the complexity of this is going to grow.  We don't want to break users because we outgrow `argparse`.

`typer` is built using `click`.  https://typer.tiangolo.com/alternatives/#click

Reply:
Switched to using click.